action.yml

```diff
@@ -20,9 +20,27 @@ inputs:
 runs:
   using: 'composite'
   steps:
+    - name: 'Determine version'
+      id: jbangversion
+      shell: bash
+      run: |
+        if [ "${{ inputs.version }}" = "latest" ]; then
+          version=$(curl -fsSL https://www.jbang.dev/releases/latest/download/version.txt)
+        else
+          version="${{ inputs.version }}"
+        fi
+        echo "VERSION=$version" >> "$GITHUB_OUTPUT"
+    - name: Cache JBang
+      id: cachejbang
+      uses: actions/cache@v4
+      with:
+        path: ~/.jbang
+        key: jbang-${{ steps.jbangversion.outputs.VERSION }}
```

Comment:
why use jbang version for the cache key?

shouldn't it be what files that changes? i.e. **/*.java files? 

and that seems like something that hsold be configurable as it is context depnendent?

Reply:
I understand JBang as follows:

Fills ~/.jbang dependent on the **./*.java files.

Thus, if A.java is changed, ~/.jbang changes.

GitHub cache updates the cache if the directory changes. Thus, a change in A.java leads to a change in ~/.jbang, which leads to an update of the cache.

---

I assume that JBang's inner workings may change from version to version. To be sure, a new cache is setup for each version.

This is also a kind of hack to clean up old `-SNAPSHOT` downloads.

A more deterministic approach could be to use the YYYY-mm as cache key to enable a monthly refresh of depdencies.